rgdc/rgdc/session.py

```diff
@@ -24,3 +26,24 @@ def __init__(self, base_url: str, auth_header: Optional[str] = None):
 
         if auth_header:
             self.headers['Authorization'] = auth_header
+
+
+def retry_rgdcsession(*args, retries: Optional[int] = 5, **kwargs):
+    """
+    Initialize a session with a ResonantGeoData server with automatic retries.
+
+    See RgdcSession for args.
+
+    References:
+        https://www.peterbe.com/plog/best-practice-with-retries-with-requests
+        https://findwork.dev/blog/advanced-usage-python-requests-timeouts-retries-hooks/
+    """
+    session = RgdcSession(*args, **kwargs)
+    retry = Retry(
+        total=retries, status_forcelist=[429, 503], method_whitelist=['GET'], backoff_factor=1
+    )
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('https://', adapter)
+    session.mount('http://', adapter)
+
+    return session
```

Comment:
I agree this would be cleaner. I was worried that trying to `self.mount` before `__init__()` had exited may lead to problems. I'll try it out and see what happens...

Reply:
seems ok!